### DISCUSSION TITLE
How to deploy a development cluster with storage?

### DISCUSSION THREAD

**Discussion by dragonly** [databendlabs/databend#3120] [Q&A] [answered]
Currently there seems no docs about how to deploy a working cluster for development/hacking with storage?
I only encountered `make run` in the documentation, which only spins up a single `databend-meta` and a `databend-query` process.

**Comment by ZhiHanZ** [comment, marked as answer]
Hi @dragonly , currently we support local and S3 storage backend.
To run with local storage, you could take a look at our stateful test script https://github.com/datafuselabs/databend/blob/main/scripts/ci/ci-run-stateful-tests-cluster.sh.
To run with S3 storage, you could look at https://github.com/datafuselabs/databend/blob/main/scripts/ci/ci-run-stateful-tests-standalone-s3.sh, bendctl integration with S3 storage is WIP, and welcome to contribute

**Reply by dragonly** [reply]
Thanks for your reply!
I'll take a look at them.
